### ISSUE TITLE
Specify *exact* versions to use in installation

### ISSUE THREAD

**Issue by joshmouch** [Unity-Technologies/ml-agents#5987]
I'm trying to follow the installation document, and no matter what combinations of versions I use, I run into one versioning problem or another.

The `installation.md` has wording like: "we recommend installing Python 3.7".  But yet a few paragraphs earlier or even in the immediately preceding sentence it says, "Install Python (3.10.12 or higher)".
If I try version 3.7 then I can't use the mlagents 0.30.0 that is instructed to use later on in the documentation because 0.29.0 is the latest version that supports Python 3.7.
If I try the latest python (3.12), then I have to use PyTorch 2.x, which doesn't seem to work.
I've tried 3.8, 3.9, and 3.10.

I've also noticed that mlagent 30.0 needs <= python 3.10.8, but the latest security release for 3.10 is 3.10.12, so that's why that version isn't working.

Using latest 3.10, I have to use magent 28, but then I eventually get an error about protobuf version being too high.

I've done all of this within virtual Python environments, which the documentation implies resolves dependency issues, but I'm not having much luck with that.

The point here being: if the versioning on all these libraries is that brittle, then the installation documentation needs be _very_ exact on what versions to use of every component.  As it is, if you follow the documentation, you can't get a working installation.
Or maybe, is it possible to have a single command to run that installs all of the dependencies with their required versions?

**Comment by miguelalonsojr** [comment]
I've updated the installation instructions. Give it a shot: https://unity-technologies.github.io/ml-agents/Installation/

TL; DR: install from the repo.

**Comment by miguelalonsojr** [comment]
Closing as this is not a bug.
